src/FixerConfiguration/FixerOption.php

```diff
@@ -0,0 +1,225 @@
+ */
+
+namespace PhpCsFixer\FixerConfiguration;
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+
+final class FixerOption implements FixerOptionInterface
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $description;
+
+    /**
+     * @var mixed
+     */
+    private $default;
+
+    /**
+     * @var bool
+     */
+    private $useDefault = false;
+
+    /**
+     * @var array|null
+     */
+    private $allowedTypes;
+
+    /**
+     * @var array|null
+     */
+    private $allowedValues;
+
+    /**
+     * @var \Closure|null
+     */
+    private $normalizer;
+
+    /**
+     * @param string $name
+     * @param string $description
+     */
+    public function __construct($name, $description)
+    {
+        $this->name = $name;
+        $this->description = $description;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return $this->description;
+    }
+
+    /**
+     * @param mixed $default
+     *
+     * @return $this
+     */
+    public function setDefault($default)
+    {
+        $this->default = $default;
+        $this->useDefault = true;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasDefault()
+    {
+        return $this->useDefault;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefault()
+    {
+        if (!$this->hasDefault()) {
+            throw new \LogicException('No default value defined.');
+        }
+
+        return $this->default;
+    }
+
+    /**
+     * @param string[]|string|null $allowedTypes
+     *
+     * @return $this
+     */
+    public function setAllowedTypes($allowedTypes)
+    {
+        if (null !== $allowedTypes && !is_array($allowedTypes)) {
+            $allowedTypes = array($allowedTypes);
+        }
+
+        $this->allowedTypes = $allowedTypes;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAllowedTypes()
+    {
+        return $this->allowedTypes;
+    }
+
+    /**
+     * @param string[]|string|null $allowedValues
+     *
+     * @return $this
+     */
+    public function setAllowedValues($allowedValues)
+    {
+        if (!is_array($allowedValues)) {
+            $allowedValues = array($allowedValues);
+        }
+
+        $this->allowedValues = $allowedValues;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAllowedValues()
+    {
+        return $this->allowedValues;
+    }
+
+    /**
+     * Sets the given option to only accept an array with a subset of the given values.
+     *
+     * @param array $allowedArrayValues
+     *
+     * @return $this
+     */
+    public function setAllowedValueIsSubsetOf(array $allowedArrayValues)
```

Comment:
Sorry, I would totally drop it.
This method breaks so many things. It's a convenient helper, but it's nothing more. It doesn't provide any functionallity.
And it provides issues.

There are 2 concepts in `OptionsResolver` component:
- normalizers - which **modify** value (eg ensuring there is no trailing spaces in value)
- validators (`setAllowed...`) - which **check** value (eg if they are in set or passing function)

If I would like to set up few validators:
```php
$foo->setAllowedValues([
    [$array, $of, $allowed, $values],
    function ($value) { /* validates value */ },
]);
$foo->setAllowedValueIsSubsetOf([...]);
```

what is the effect? first two validators were lost with 3rd one.

One could try to collect all validators, but then those methods wouldn't be setters, they would be adders. I don't like that.
IMO, `setAllowedValueIsSubsetOf` should be removed.